doc/hacking.rst

```diff
@@ -446,6 +446,10 @@ Good:
   ensures that the pattern matching will break when a field is added. To ignore
   a field, add ``; d = _``, not ``; _``.
 
+- When ignoring the value of a let binding ``let _ = ...``  we add type
+  annotations to the ignored value ``let (_ : t) = ...``. This is to avoid
+  accidentally changing the type of the ignored value.
```

Comment:
I don't know what it means to accidentally change the type of the ignored value.

We do this convention because:
* Make sure we never ignore `Fiber.t` accidentally. Functions that return `Fiber.t` always free of side effects so we need to bind on the result to force the side effect.
* Whenever a function is changed to return an error via its return value, we want the compiler to notify all the callers that need to be updated.

Reply:
Updated. I essentially meant your second point.